internal/github/github_test.go

```diff
@@ -0,0 +1,874 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package github
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-github/v69/github"
+)
+
+func newTestServerAndClient(t *testing.T, handler http.HandlerFunc) (*Client, *httptest.Server) {
+	t.Helper()
+	server := httptest.NewServer(handler)
+	client, err := NewClient("test-token")
+	if err != nil {
+		t.Fatalf("NewClient() error = %v", err)
+	}
+	c := github.NewClient(server.Client())
+	c.BaseURL, _ = url.Parse(server.URL + "/")
+	client.Client = c
+	return client, server
+}
+
+func TestCreatePullRequest(t *testing.T) {
+	repo := &Repository{Owner: "test-owner", Name: "test-repo"}
+	remoteBranch := "test-branch"
+	title := "Test PR"
+	cases := []struct {
```

Comment:
Can you please re-structure the test according to https://github.com/googleapis/librarian/blob/main/doc/howwewritego.md#table-driven-tests?

Same for tests below